x/tokenfactory/keeper/before_send_test.go

```diff
@@ -253,36 +253,69 @@ func (suite *KeeperTestSuite) TestNonWhitelistedHooksNotCalled() {
-			useFactoryDenom: true,
+			blockBeforeSend: false,
+			useHookedDenom:  true,
+			gasLimit:        30_000_000,
+			expectedError:   false,
+			expectedPanic:   false,
 		},
 		{
-			name:            "sending tokenfactory denom from account to account with infinite contract should panic",
-			wasmFile:        "./testdata/infinite_track_beforesend.wasm",
-			useFactoryDenom: true,
+			name:            "sending tokenfactory denom with infinite contract should not return error on blockBeforeSend",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm", // https://github.com/neutron-org/neutron-dev-contracts/tree/chore/additional-tf-test-contracts/contracts/infinite-track-beforesend
 			blockBeforeSend: true,
+			useHookedDenom:  true,
+			gasLimit:        30_000_000,
+			expectedError:   true,
+			expectedPanic:   false,
+		},
+		{
+			name:            "block_before_send: sending tokenfactory denom from module to module with infinite contract should panic when outer layer gas is breached on trackBeforeSend",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm", // https://github.com/neutron-org/neutron-dev-contracts/tree/chore/additional-tf-test-contracts/contracts/infinite-track-beforesend
+			blockBeforeSend: false,
+			useHookedDenom:  true,
+			gasLimit:        300_000, // lower than 500_000 inner constant, so it should trigger outer context outOfGas panic
+			expectedError:   false,
+			expectedPanic:   true,
 		},
 		{
-			name:            "sending non-tokenfactory denom from module to module with infinite contract should not panic",
-			wasmFile:        "./testdata/infinite_track_beforesend.wasm",
-			tokenToSend:     sdk.NewCoins(sdk.NewInt64Coin("foo", 1000000)),
-			useFactoryDenom: false,
+			name:            "block_before_send: sending tokenfactory denom with infinite contract should panic when outer layer gas is breached on blockBeforeSend",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm", // https://github.com/neutron-org/neutron-dev-contracts/tree/chore/additional-tf-test-contracts/contracts/infinite-track-beforesend
+			blockBeforeSend: true,
+			useHookedDenom:  true,
+			gasLimit:        300_000, // lower than 500_000 inner constant, so it should trigger outer context outOfGas panic
+			expectedError:   false,
+			expectedPanic:   true,
 		},
 		{
-			name:            "Try using no 100",
+			name:            "sending non subscribed denom from module to module with infinite contract should not panic or return error",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm", // https://github.com/neutron-org/neutron-dev-contracts/tree/chore/additional-tf-test-contracts/contracts/infinite-track-beforesend
+			useHookedDenom:  false,
+			blockBeforeSend: false,
+			gasLimit:        30_000_000,
+			expectedError:   false,
+			expectedPanic:   false,
+		},
+		{
+			name:            "Try using no 100 contract - sends that more then 100 gets denied",
```

Comment:
don't like the description: it's a bit incorrect in grammar and doesn't align with the style of other test case names. here's an idea from my side. if you don't like it I'd like to know your thoughts
```suggestion
			name:            "sending tokenfactory denom with expected error from contract",
```